### ISSUE TITLE
PLW0120 (`else` clause on loop without a `break` statement) false positive inside a `match`

### ISSUE THREAD

**Issue by bluetech** [charliermarsh/ruff#3131]
Excited by the new `match` support, I ran it on a big project which contains a lot of match statements. The following is the only issue.

Snippet:

```py
while 1:
    match 10:
        case _:
            break
else:
    print('here')
```

Command: `ruff --isolated --select PLW0120 x.py`

Expected: no violations -- the `else` is definitely not redundant here.

Actual:

```
x.py:5:1: PLW0120 `else` clause on loop without a `break` statement; remove the `else` and de-indent all the code inside it
```

Version: 0.0.251

**Comment by charliermarsh** [comment]
Thanks, will fix today.
